src/units/zone_signer.rs

```diff
@@ -469,7 +469,9 @@ impl ZoneSigner {
 
             // Do NOT sign a zone that is halted.
             if zone_state.pipeline_mode != PipelineMode::Running {
-                return Err(SignerError::PipelineIsHalted);
+                // TODO: This accidentally sets an existing soft-halt to a hard-halt.
+                // return Err(SignerError::PipelineIsHalted);
+                return Ok(());
```

Comment:
If you return Ok here the zone will end up in zone status output as having finished signing rather than being aborted. The alternative is to add an error variant like `CannotResignNonPublishedZone` and handle it specially in `handle_command()`.